book/03-git-branching/sections/rebasing.asc

```diff
@@ -228,8 +228,10 @@ So what if there was a messy series of merge commits?
 That's how it happened, and the repository should preserve that for posterity.
 
 The opposing point of view is that the commit history is the *story of how your project was made.*
-You wouldn't publish the first draft of a book, and the manual for how to maintain your software deserves careful editing.
-This is the camp that uses tools like `rebase` and `filter-branch` to tell the story in the way that's best for future readers.
+You wouldn't publish the first draft of a book, so why show your messy work?
+Remember, you are likely to need this history yourself, when fixing a bug, or figuring out why you made a particular change.
+People in this camp clean up their commit(s) and commit messages, before the work is merged/rebased into the mainline branch.
```

Comment:
Here's an alternate version, what do you think?

```suggestion
When you're working on a project, you may need a record of all your missteps and dead-end paths, but when it's time to show your work to the world, you may want to tell a more coherent story of how to get from A to B.
People in this camp use tools like rebase and filter-branch to rewrite their commits before they're merged into the mainline branch.
```

Reply:
@